cmd/snap-update-ns/bootstrap.c

```diff
@@ -486,14 +483,15 @@ void bootstrap(int argc, char **argv, char **envp)
 	const char *snap_name = NULL;
 	bool should_setns = false;
 	bool process_user_fstab = false;
-	unsigned long uid = 0;
+	unsigned long uid = getuid();
 	process_arguments(argc, argv, &snap_name, &should_setns,
 			  &process_user_fstab, &uid);
-	if (process_user_fstab) {
-		switch_to_privileged_user();
-		// switch_to_privileged_user sets bootstrap_{errno,msg}
-	} else if (snap_name != NULL && should_setns) {
-		setns_into_snap(snap_name);
+	if (should_setns && snap_name != NULL) {
```

Comment:
Perhaps add a comment here:
```
// If needed, enter the snap's namespace using either the user fstab with our uid
// or the system wide snap fstab with uid 0
```

Reply:
A snap should always be set, no?  it would seem clearer if the snap_name != NULL would guard any action in the last part of bootstrap